### DISCUSSION TITLE
Making an equality filter mandotary (requiring id to filter)

### DISCUSSION THREAD

**Discussion by silenceofthewans** [pocketbase/pocketbase#2844] [Q&A] [answered]
I have an accounts array which can be empty if there are no accounts yet and as filter is optional the only way I could find to not return all records was to send an empty string to the field if array is empty.  Is following the right way for now or is there a better approach?
```TS
const accountsId = accounts.map((a) => `account="${a.id}"`).join('||') || 'account=""';

const response = await db.collection("transactions").getList(1, 20, {
  sort: '-created',
  filter: accountsId
});
```

**Comment by ganigeorgiev** [comment, marked as answer]
You can also try with the `:length` modifier - `account:length=0`, but in your case an equality with empty string (or `!="[]"`) may perform better better in large datasets.
